### PR TITLE
[Impeller] migrate texture fill shaders to half precision.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Work towards https://github.com/flutter/flutter/issues/115044

### CODE REVIEW

**Review by dnfield — APPROVED** [review]

File: impeller/compiler/shader_lib/impeller/texture.glsl [hunk 1]
```diff
@@ -92,6 +93,26 @@ vec4 IPSampleWithTileMode(sampler2D tex,
   return texture(tex, coords);
 }
 
+const float16_t kTileModeDecalHf = 3.0hf;
+
+/// Sample a texture, emulating a specific tile mode.
+///
+/// This is useful for Impeller graphics backend that don't have native support
+/// for Decal.
+f16vec4 IPSampleWithTileModeHf(f16sampler2D tex,
```
Comment (bdero) [inline impeller/compiler/shader_lib/impeller/texture.glsl:102]: Maybe `IPHalfSampleWithTileMode` to follow the pattern of putting the type info near the beginning? Not sure the best route to go for this, but this is public and available to Flutter users as well. We should probably document some kind of convention for these at some point.

**Review by bdero — APPROVED** [review]
LGTM

File: impeller/compiler/shader_lib/impeller/texture.glsl (hunk 1 above)
Comment (jonahwilliams) [reply impeller/compiler/shader_lib/impeller/texture.glsl:102]: I thought we only exported runtime effect?

Comment (jonahwilliams) [reply impeller/compiler/shader_lib/impeller/texture.glsl]: (Renaming done)

Comment (jonahwilliams) [reply impeller/compiler/shader_lib/impeller/texture.glsl]: At a certain point though, we'll probably drop this and only have a half precision verison.

**Pull request merged**